packages/xstate-vue/src/useFsm.ts

```diff
@@ -0,0 +1,26 @@
+import { ref, onBeforeMount, onBeforeUnmount, Ref } from '@vue/composition-api';
+import { StateMachine, EventObject, interpret } from '@xstate/fsm';
+import { AnyEventObject } from 'xstate';
+
+export function useFsm<TC, TE extends EventObject = AnyEventObject>(
+  stateMachine: StateMachine.Machine<TC, TE, any>
+): {
+  state: Ref<StateMachine.State<TC, TE, any>>;
+  send: StateMachine.Service<TC, TE>['send'];
+  service: Ref<StateMachine.Service<TC, TE>>;
+} {
+  const state = ref<StateMachine.State<TC, TE, any>>(stateMachine.initialState);
```

Comment:
do you need refs for those? i might miss something about vue, but looking at this code from React's perspective - it seems like this wouldn't be needed. In Vue this whole hook gets executed only once (from what I understand). Are refs different in Vue from what we have in ReacT? Could you describe how they work in Vue?

Reply:
In the composition API every reactive value you want to have, you wrap it with either a `const count = ref(0)` or you create a `const state = reactive({ count: 1 }`. If you want to mutate a ref's value you'd have to use its `value` property. So mutating the `count` to 1 would be `coult.value = 1`

As we want the state to be a reactive property I wrap the given `initalState` so I can later mutate it by executing `subscribe(s => (state.value = s))`. 

At the end of a `setup()` function you return those refs, so you can use them in your components as reactive values, methods etc.

Also see the [API documentation for ref](https://vue-composition-api-rfc.netlify.com/api.html#ref)

Reply:
Right, a Vue `ref` is different than a React `ref` from my understanding.

Reply:
Ok - so I've read the explanation and the docs and I think I understand Vue's ref and reactive somewhat OK.

So given that - I understand why `state` is a ref here, but I don't get why `service` is one too. It's only initialized and you never write to it again, so it should be just a plain value.